packages/nouns-contracts/contracts/governance/NounsDAOV3Proposals.sol

```diff
@@ -256,6 +258,70 @@ library NounsDAOV3Proposals {
             updateMessage
         );
     }
+    
+    function updateProposalTransactions(
+        NounsDAOStorageV3.StorageV3 storage ds,
+        uint256 proposalId,
+        address[] memory targets,
+        uint256[] memory values,
+        string[] memory signatures,
+        bytes[] memory calldatas,
+        string memory updateMessage
+    ) external {
+        updateProposalTransactionsInternal(ds, proposalId, targets, values, signatures, calldatas);
+
+        emit ProposalTransactionsUpdated(
+            proposalId,
+            msg.sender,
+            targets,
+            values,
+            signatures,
+            calldatas,
+            updateMessage
+        );
+    }
+
+    function updateProposalTransactionsInternal(
+        NounsDAOStorageV3.StorageV3 storage ds,
+        uint256 proposalId,
+        address[] memory targets,
+        uint256[] memory values,
+        string[] memory signatures,
+        bytes[] memory calldatas
+    ) internal {
+        checkProposalTxs(ProposalTxs(targets, values, signatures, calldatas));
+
+        NounsDAOStorageV3.Proposal storage proposal = ds._proposals[proposalId];
+        checkProposalUpdatable(ds, proposalId, proposal);
+
+        proposal.targets = targets;
+        proposal.values = values;
+        proposal.signatures = signatures;
+        proposal.calldatas = calldatas;
+    }
+
+    function updateProposalDescription(
+        NounsDAOStorageV3.StorageV3 storage ds,
+        uint256 proposalId,
+        string calldata description,
+        string calldata updateMessage
+    ) external {
+        NounsDAOStorageV3.Proposal storage proposal = ds._proposals[proposalId];
+        checkProposalUpdatable(ds, proposalId, proposal);
+
+        emit ProposalDescriptionUpdated(proposalId, msg.sender, description, updateMessage);
+    }
+
+    function checkProposalUpdatable(
+        NounsDAOStorageV3.StorageV3 storage ds,
+        uint256 proposalId,
+        NounsDAOStorageV3.Proposal storage proposal
+    ) internal view {
+        // TODO: gas: does reading the proposal once save gas?
+        if (state(ds, proposalId) != NounsDAOStorageV3.ProposalState.Updatable) revert CanOnlyEditUpdatableProposals();
```

Comment:
@davidbrai maybe we should refactor state such that there's a version of it that takes a `Proposal storage proposal` parameter so we avoid the double reading we perform when we're holding a proposal in a function and then re-reading it when calling state?

Reply:
I see the related todo you added as a comment above :) I think it's probably not very hard for you to look into it as you're writing some unit tests for these functions?